File: 1381.设计一个支持增量操作的栈.go
/*
 * @lc app=leetcode.cn id=1381 lang=golang
 *
 * [1381] 设计一个支持增量操作的栈
 */

// @lc code=start
type CustomStack struct {
	Stack   []int
	MaxSize int
}

func Constructor(maxSize int) CustomStack {
	return CustomStack{
		Stack:   make([]int, 0, maxSize),
		MaxSize: 0,
	}

}

func (this *CustomStack) Push(x int) {
	if this.MaxSize < cap(this.Stack) {
		this.Stack = append(this.Stack, x)
		this.MaxSize++
	}

}

func (this *CustomStack) Pop() int {
	if this.MaxSize == 0 {
		return -1
	}
	fmt.Println(this.Stack)
	top := this.Stack[this.MaxSize-1]
	this.Stack = this.Stack[:this.MaxSize-1]
	this.MaxSize--
	return top

}

func (this *CustomStack) Increment(k int, val int) {
	if k > this.MaxSize {
		k = this.MaxSize
	}
	for i := 0; i < k; i++ {
		this.Stack[i] += val
	}

}

/**
 * Your CustomStack object will be instantiated and called as such:
 * obj := Constructor(maxSize);
 * obj.Push(x);
 * param_2 := obj.Pop();
 * obj.Increment(k,val);
 */
// @lc code=end

File: 1381.设计一个支持增量操作的栈.py
#
# @lc app=leetcode.cn id=1381 lang=python3
#
# [1381] 设计一个支持增量操作的栈
#

# @lc code=start
class CustomStack:

    def __init__(self, maxSize: int):
        self.maxSize = maxSize
        self.stack = []


    def push(self, x: int) -> None:
        if len(self.stack) < self.maxSize:
            self.stack.append(x)


    def pop(self) -> int:
        if self.stack:
            return self.stack.pop()
        return -1


    def increment(self, k: int, val: int) -> None:
        for i in range(min(k, len(self.stack))):
            self.stack[i] += val



# Your CustomStack object will be instantiated and called as such:
# obj = CustomStack(maxSize)
# obj.push(x)
# param_2 = obj.pop()
# obj.increment(k,val)
# @lc code=end

